4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AnimalShelterCode
+{
+    public class AnimalShelter
+    {
+        public int AnimalCounter { get; private set; }
+
+        public List<Animal> Animals { get; private set; }
+
+        public AnimalShelter()
+        {
+            Animals = new List<Animal>();
+            AnimalCounter = 1;
+        }
+
+        public OperationResult AddAnimal(Animal animal)
+        {
+            if (animal.Type == AnimalType.Bear || animal.Type == AnimalType.Deer) 
+            { 
+                return new OperationResult($"Animal Type {animal.Type} is not Supported");
+            }
+            else
+            {
+                animal.UniqueAnimalId = AnimalCounter++;
+                Animals.Add(animal);
+                return new OperationResult(animal);
+            }
+        }
+
+        public List<Animal> GetAnimals(AnimalType filtertype)
+        {
+            var filteredlist = from animal in Animals
+                               where animal.Type == filtertype
+                               select animal;
+
+            return filteredlist.ToList();
+        }
```

Comment:
This method is actually missing a rather important filter ability that will actually cause you to re-work your code. The requirements states:

- Animals that can fly
- Animals that are cats
- Animals that are dogs

Note that there are *two* "types" of requirements. What an animal "IS" and what an animal "HAS". If this were a real application, this would be a defect & classified as a missed requirement.